### ISSUE TITLE
AMI Name is random in 1.4.0/1.4.1

### ISSUE THREAD

**Issue by mlasevich** [hashicorp/packer#7779]
Using amazon-ebs with specified "ami_name" does not work under 1.4.1(or 1.4.0). Creates AMI with a random 7 character string. Same exact setup works fine in 1.3.5

Here is an sample affected packer file:
```
{
  "variables": {
      "AMI_NAME": "test-ami-{{ timestamp }}",
      "region": "{{env `PACKER_AWS_REGION`}}",
      "subnet_id": "{{env `PACKER_SUBNET`}}",
      "vpc_id": "{{env `PACKER_VPC`}}"
  },
  "builders": [{
    "type": "amazon-ebs",
    "region": "{{user `region`}}",
    "source_ami": "ami-0375ca3842950ade6",
    "instance_type": "t3.micro",
    "ssh_username": "ubuntu",
    "ami_name": "{{user `AMI_NAME`}}",
    "ami_description": "Test AMI",
    "subnet_id": "{{user `subnet_id`}}",
    "vpc_id": "{{user `vpc_id`}}",
    "encrypt_boot": "false"
  }]
}
```
and resulting output:
```
amazon-ebs output will be in this color.

==> amazon-ebs: Prevalidating AMI Name: test-ami-1561089972
    amazon-ebs: Found Image ID: ami-0375ca3842950ade6
==> amazon-ebs: Creating temporary keypair: packer_5d0c57b4-9c90-d4aa-19c9-377c0ac6d207
==> amazon-ebs: Creating temporary security group for this instance: packer_5d0c57b5-024a-e0cf-1189-9a07460ee308
==> amazon-ebs: Authorizing access to port 22 from [0.0.0.0/0] in the temporary security groups...
==> amazon-ebs: Launching a source AWS instance...
...
    amazon-ebs: Stopping instance
==> amazon-ebs: Waiting for the instance to stop...
==> amazon-ebs: Creating AMI ESNRUnG from instance i-0d6ee97437627f07d
    amazon-ebs: AMI: ami-063c60fb75bd8972e
==> amazon-ebs: Waiting for AMI to become ready...
==> amazon-ebs: Modifying attributes on AMI (ami-063c60fb75bd8972e)...
    amazon-ebs: Modifying: description
==> amazon-ebs: Modifying attributes on snapshot (snap-03653783bab9bc643)...
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: Cleaning up any extra volumes...
==> amazon-ebs: No volumes to clean up, skipping
==> amazon-ebs: Deleting temporary security group...
==> amazon-ebs: Deleting temporary keypair...
Build 'amazon-ebs' finished.

==> Builds finished. The artifacts of successful builds are:
--> amazon-ebs: AMIs were created:
us-west-2: ami-063c60fb75bd8972e
```
Interesting bit is that if I remove `"encrypt_boot": "false"`, it seems to work fine.

**Comment by azr** [comment]
Hello there, thanks for opening, what you are seeing here in the logs (`ami-063c60fb75bd8972e`), is your new AMI ID, your ami will still be named as you wanted and you can check this in the aws console.

I just tried this and it worked.

___
More:

After PR #6787 which was merged for 1.4.0; explicitly setting `encrypt_boot` to anything will have packer create a temporary image in order to copy the image to your new encrypted setting.

If the original image is not encrypted and you would like your new image to remain unencrypted, not setting `encrypt_boot` should be the way to go as there is one less copy step.

I hope that helps.

**Comment by mlasevich** [comment]
To be clear, I know what ami id is. The issue here is not ami id, but AMI name, which in this case results in random characters - despite what is being set. This is not the expected behavior. I get what you said about it being copied after creation to set the right name, but in my experience that does not happen in most cases (seems to happen sometimes, but not most of the time)

I guess I didn't read the docs enough to realize that the `encrypt_boot` is not a "boolean" actually but a ternery value. Is there any way to keep the entry to set it to the "unset" value? Can I provide a blank string or something?  We are using the packer in a parameterized way,  where same template may be used in and I would rather not have to either have multiple json files or have to manipulate json on the fly to allow optional encryption, given that setting it to false does not produce expected results

**Comment by mlasevich** [comment]
demonstration using the above packer file:
```
$ cat ./BrokenAMIName.json 
{
  "variables": {
      "AMI_NAME": "test-ami-{{ timestamp }}",
      "region": "{{env `PACKER_AWS_REGION`}}",
      "subnet_id": "{{env `PACKER_SUBNET`}}",
      "vpc_id": "{{env `PACKER_VPC`}}"
  },
  "builders": [{
    "type": "amazon-ebs",
    "region": "{{user `region`}}",
    "source_ami": "ami-0375ca3842950ade6",
    "instance_type": "t3.micro",
    "ssh_username": "ubuntu",
    "ami_name": "{{user `AMI_NAME`}}",
    "ami_description": "Test AMI",
    "subnet_id": "{{user `subnet_id`}}",
    "vpc_id": "{{user `vpc_id`}}",
    "encrypt_boot": "false"
  }]
}
$ ~/bin/packer --version
1.4.1
$ ~/bin/packer build -var 'region=us-west-2' -var "vpc_id=${VPC_ID}" -var "subnet_id=${SUBNET_ID}" ./BrokenAMIName.json 
amazon-ebs output will be in this color.

==> amazon-ebs: Prevalidating AMI Name: test-ami-1561133545
    amazon-ebs: Found Image ID: ami-0375ca3842950ade6
==> amazon-ebs: Creating temporary keypair: packer_5d0d01e9-a082-ec37-d5a3-50c2ab3aaba3
==> amazon-ebs: Creating temporary security group for this instance: packer_5d0d01eb-394d-9705-672b-8c96cb43fd7b
==> amazon-ebs: Authorizing access to port 22 from [0.0.0.0/0] in the temporary security groups...
==> amazon-ebs: Launching a source AWS instance...
==> amazon-ebs: Adding tags to source instance
    amazon-ebs: Adding tag: "Name": "Packer Builder"
    amazon-ebs: Instance ID: i-067b85e19d678737d
==> amazon-ebs: Waiting for instance (i-067b85e19d678737d) to become ready...
==> amazon-ebs: Using ssh communicator to connect: 10.182.250.39
==> amazon-ebs: Waiting for SSH to become available...
==> amazon-ebs: Connected to SSH!
==> amazon-ebs: Stopping the source instance...
    amazon-ebs: Stopping instance
==> amazon-ebs: Waiting for the instance to stop...
==> amazon-ebs: Creating AMI lBnarnJ from instance i-067b85e19d678737d
    amazon-ebs: AMI: ami-00b852db83163b8b2
==> amazon-ebs: Waiting for AMI to become ready...
==> amazon-ebs: Modifying attributes on AMI (ami-00b852db83163b8b2)...
    amazon-ebs: Modifying: description
==> amazon-ebs: Modifying attributes on snapshot (snap-05f74f589ee6e8672)...
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: Cleaning up any extra volumes...
==> amazon-ebs: No volumes to clean up, skipping
==> amazon-ebs: Deleting temporary security group...
==> amazon-ebs: Deleting temporary keypair...
Build 'amazon-ebs' finished.

==> Builds finished. The artifacts of successful builds are:
--> amazon-ebs: AMIs were created:
us-west-2: ami-00b852db83163b8b2

$ aws ec2  describe-images --query 'Images[*].[ImageId,Name]' --image-ids ami-00b852db83163b8b2
ami-00b852db83163b8b2	lBnarnJ
```
As you see, no secondary copy and the AMI name is not set correctly

**Comment by SwampDragons** [comment]
@mlasevich sorry about this! The good news is that it's a duplicate of https://github.com/hashicorp/packer/issues/7689 and has already been fixed on the master branch. It'll be fixed in the 1.4.2 release. It's fixed on the master branch.
